File: implementations/react-native/design-patterns-rn/src/patterns/behavioral/chain_of_responsibility/escalamiento_gerencia.ts
import { SupportHandler } from './base_handler';
import type { HandleResult, Ticket } from './types';

export class EscalamientoGerencia extends SupportHandler {
  handle(_req: Ticket, trail: string[] = []): HandleResult {
    this.push(trail);
    return { handledBy: 'EscalamientoGerencia', message: 'Escalado al Gerente de Guardia.', trail };
  }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/chain_of_responsibility/escalamiento_gerencia.dart
import 'handler.dart';
import 'types.dart';

class EscalamientoGerencia extends Handler {
  @override
  HandleResult handle(Ticket req, {List<String>? trail}) {
    final t = trail ?? <String>[];
    push(t);
    return HandleResult(
      handledBy: 'EscalamientoGerencia',
      message: 'Escalado al Gerente de Guardia.',
      trail: t,
    );
  }
}
